src/freadR.c

```diff
@@ -126,7 +126,7 @@ SEXP freadR(
   if (isReal(nrowLimitArg)) {
     if (R_FINITE(REAL(nrowLimitArg)[0]) && REAL(nrowLimitArg)[0]>=0.0) args.nrowLimit = (int64_t)(REAL(nrowLimitArg)[0]);
   } else {
-    if (INTEGER(nrowLimitArg)[0]>=1) args.nrowLimit = (int64_t)INTEGER(nrowLimitArg)[0];
+    if (INTEGER(nrowLimitArg)[0]>=0) args.nrowLimit = (int64_t)INTEGER(nrowLimitArg)[0];
```

Comment:
I'm glad it works but I'm not sure why `nrows=0` would work differently than `nrows=0L`...

Reply:
Actually thats quite interesting. Apparently `0L` is not real and hence the second case triggers.

See also
```
library(inline)
is_real <- cfunction(c("x" = "ANY"), "
  return ScalarLogical(isReal(x));
")

is_real(0) #returns TRUE
is_real(0L) #returns FALSE
```

Reply:
Hmm and we use real here because we allow `nrows=Inf` to mean "everything". What about just doing `as.numeric(nrows)` in the `fread.R` code?

Reply:
Would also be a possibility. But then we could also remove the else branch in `fread.c` since it should not be reachable anymore?

Reply:
Try it out and see if we pass tests. There might be some other use case we're not thinking of

Reply:
Passes all tests.